src/linear_algebra/affine_space/independent.lean

```diff
@@ -239,6 +281,47 @@ by simp [ha]
 
 end affine_independent
 
+section field
+
+variables {k : Type*} {V : Type*} {P : Type*} [field k] [add_comm_group V] [module k V]
+variables [affine_space V P]
+include V
+
+/-- An affinely independent set of points can be extended to such a
+set that spans the whole space. -/
+lemma exists_subset_affine_independent_affine_span_eq_top {s : set P}
+    (h : affine_independent k (λ p, p : s → P)) :
+  ∃ t : set P, s ⊆ t ∧ affine_independent k (λ p, p : t → P) ∧ affine_span k t = ⊤ :=
+begin
+  by_cases hn : s.nonempty,
```

Comment:
In this situation I think you almost always want to use `rcases s.eq_empty_or_nonempty with rfl | ⟨p1, hp1⟩`. Of course you'll need to swap the two subproofs and delete a couple of lines. I would also prefer if you could use subscripts instead of `p1` and `hp1`.

Reply:
Made those changes.